clients/src/main/java/org/apache/kafka/common/serialization/ListDeserializer.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.serialization;
+
+import java.io.ByteArrayInputStream;
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.lang.reflect.Constructor;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ListDeserializer<T> implements Deserializer<List<T>> {
+
+    private final Class listClass;
+    private final Deserializer<T> deserializer;
+    private final Integer primitiveSize;
+
+    private Map<Class, Integer> primitiveDeserializers = Stream.of(new Object[][]{
```

Comment:
I using `Stream.of` the best was to populate the Map? Seems to be unnecessarily complex to me?

Reply:
Replaced it with simpler approach:
```
new HashMap<Class, Integer>() {{
            put(LongDeserializer.class, 8);
            ...
    }};
```